config/src/config/mod.rs

```diff
@@ -327,8 +326,14 @@ impl NodeConfig {
 
             let validator_network = self.validator_network.as_mut().unwrap();
             validator_network.random_with_peer_id(rng, Some(peer_id));
-            test.random_consensus_key(rng);
+            // We want to produce this key twice
+            let mut cloned_rng = rng.clone();
```

Comment:
nit: I'd probably inline cloned_rng. I think it's somewhat obvious that we want to produce keys based on (potentially seeded) rngs

Reply:
That would result in some slightly weird code unless we wanted to change the structure of the following code. I, personally, found this more readable and explicit. But perhaps you're suggesting something else?

Reply:
I mean the ideal state is we would have clone available -- but we do not lol